### PR TITLE
1534-review-guide-for-replacing-servers

### LINKED ISSUE #1534: Review guide for replacing servers list
We received some feedback that this guide has incorrect or outdated instructions. Test the guide and update as needed.

See: https://redocly.com/docs/cli/guides/replace-servers-url/

### PR DESCRIPTION
**Pull request by JLekawa**
Removed numbering form headers, removed "We do you do" admonition, split Prerequisites, improved language for consistency.

Fixes #1534 

## Check yourself

- [x] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/guides/replace-servers-url.md [hunk 1]
```diff
@@ -6,41 +6,48 @@ This page describes how to replace the server URL with a decorator for a given e
...
-            description: Status OK
-          '400':
-            description: Status not OK
-  ```
 - Use your favorite IDE for editing files (we use VS Code and have the [Redocly extension](../../redocly-openapi/index.md) installed).
 
-## Step 1: Create a custom plugin
+## Create a demo specification file
+
+In this step, create an OpenAPI specification file.
```
Comment (HCloward) [inline docs/guides/replace-servers-url.md:15]:
```suggestion
In this step, create an OpenAPI definition file.
```

File: docs/guides/replace-servers-url.md [hunk 2]
```diff
@@ -6,41 +6,48 @@ This page describes how to replace the server URL with a decorator for a given e
...
-        responses:
-          '204':
-            description: Status OK
-          '400':
-            description: Status not OK
-  ```
 - Use your favorite IDE for editing files (we use VS Code and have the [Redocly extension](../../redocly-openapi/index.md) installed).
 
-## Step 1: Create a custom plugin
+## Create a demo specification file
```
Comment (HCloward) [inline docs/guides/replace-servers-url.md:13]:
```suggestion
## Create a demo folder and definition file
```

File: docs/guides/replace-servers-url.md (hunk 1 above)
Comment (HCloward) [reply docs/guides/replace-servers-url.md:15]: Maybe we should say here why users are creating this folder and file. Something like, before we can create the plugin we need a sample definition document to test it with. In this step we will create the sample folder and file to use for testing our plugin later.

Comment (lornajane) [reply docs/guides/replace-servers-url.md:15]: DESCRIPTION. Not definition. (sorry for shouting, caps lock was on but I figured it sort of made my point :laughing: )

File: docs/guides/replace-servers-url.md [hunk 3]
```diff
@@ -6,41 +6,48 @@ This page describes how to replace the server URL with a decorator for a given e
...
+```
+
+3. Save the file.
+
+## Create a custom plugin
 
 In this step, create a custom plugin and define the decorator dependency.
 
-1. Create a directory called `plugins`.
+1. Inside `replace-servers-demo`, create a directory called `plugins`.
```
Comment (HCloward) [inline docs/guides/replace-servers-url.md:48]:
```suggestion
1. In the `replace-servers-demo` folder, create a directory called `plugins`.
```

File: docs/guides/replace-servers-url.md [hunk 4]
```diff
@@ -92,13 +100,17 @@ function ReplaceServersURL({serverUrl}) {
...
+If you change the name of the decorators directory, make sure to also change it in line 1 of the [`plugin.js` file](#create-a-custom-plugin).
 {% /admonition %}
 
-## Step 3: Configure the plugin for use
+## Configure the plugin for use
+
+To use the decorator, register your your `plugins` and `decorators` in the [Redocly configuration file](../configuration/index.md).
 
-To use the decorator, register your plugin in your Redocly configuration file. Register your `plugins` and `decorators`.
+- In your Redocly configuration file, add this information:
```
Comment (HCloward) [inline docs/guides/replace-servers-url.md:113]: Does the following yaml register the plugin and decorators?

File: package.json [hunk 5]
```diff
@@ -72,7 +72,7 @@
     "json-schema-to-ts": "^3.0.0",
     "null-loader": "^4.0.0",
     "outdent": "^0.7.1",
-    "prettier": "^2.1.2",
+    "prettier": "2.8.8",
```
Comment (HCloward) [inline package.json:75]: Did you mean to update the package.json file?

**Review by HCloward — COMMENTED** [review]
I love the update to the subheadings. Also, excellent work on breaking down some of these larger steps to make more digestible chunks for the reader. I think there are places where the reader can be better guided towards the ultimate goal of the doc, but otherwise this is a great improvement. Also, looks like you may need to run prettier locally to fix a couple of files.

File: package.json (hunk 5 above)
Comment (lornajane) [reply package.json:75]: I recommend you revert this change, it's possible that the updated tool version changes its behaviour and causes the prettier problems that are failing the build.

File: docs/guides/replace-servers-url.md [hunk 6]
```diff
@@ -6,41 +6,48 @@ This page describes how to replace the server URL with a decorator for a given e
...
+1. Create a new directory and name it `replace-servers-demo`.
+
+2. In `replace-servers-demo`, create an `original.yaml` file with the following content:
+
+```yaml
+openapi: 3.1.0
+    info:
+  version: 1.0.0
+  title: Custom decorators demo
+  description: The servers URL is replaced by the decorator during the `bundle` process.
```
Comment (tatomyr) [inline docs/guides/replace-servers-url.md:26]:
```suggestion
    info:
      version: 1.0.0
      title: Custom decorators demo
      description: The servers URL is replaced by the decorator during the `bundle` process.
```

**Review by lornajane — COMMENTED** [review]
This is in good shape. If you've finished working on it then mark it as "Ready for Review" so we know we should move it forward.

My (very minor) comments/suggestions:
 - consider adding subheadings for the two sections under "Verify the output"
 - I'd love another 2 sentences in the opening paragraph about the problems that could be solved by changing a server URL, basically a "why should I care?" answer.
 - this guide needs an ending! Recap what they did, why it's awesome, and offer some follow-on links, such as a blog post or the cookbook, or some more docs.

**Review by lornajane — APPROVED** [review]
BIG improvement, thanks!

**Pull request merged**